component-tools/src/test/java/org/talend/sdk/component/tools/validator/ActionValidatorTest.java

```diff
@@ -263,4 +277,41 @@ public Record guessProcessorSchemaKo6(@Option FakeDataSet configuration, RecordB
             return null;
         }
     }
+
+    @Service
+    static class ActionDynamicDependenciesOK {
+
+        @DynamicDependencies("test-all")
+        public List<String> getDynamicDependencies(@Option("configuration") final DataSet dataset) {
```

Comment:
You're cheating on the test here :-)
`org.talend.sdk.component.api.configuration.type.DataSet` is an annotation ! Same my previous comment.
Instead use the `FakeDataSet` class for accurate test.